### PR TITLE
Handle showcmd enabled

### PR DESCRIPTION
**Pull request by smhc**

### CODE REVIEW

**Comment by smhc** [comment]
This PR is to handle the wrapped echo message which causes the 'Press ENTER' prompt for diagnostics display.

The showcmd option is enabled by default in vim and neovim and reduces the width for displaying messages.
The ruler option also imposes a further restriction, but only if the status-line is disabled.

**Comment by smhc** [comment]
This is a simple, yet useful fix that turns out is similar to how it was fixed in languageclient-neovim:
https://github.com/autozimu/LanguageClient-neovim/blob/f9a9123582abfd0f0433e7f8813daf7e615a18b6/autoload/LanguageClient.vim#L20

Without this fix users are regularly interrupted with 'press enter' prompts while navigating code.

File: autoload/lsp/utils.vim [hunk 1]
```diff
@@ -142,6 +142,14 @@ endfunction
...
+    
+    if &showcmd
+        let l:winwidth -= 11
+    endif
+
+    if &laststatus != 2 && &ruler
+        let l:winwidth -= 18
+    endif
 
     if l:winwidth < strdisplaywidth(l:msg)
```
Comment (mattn) [inline autoload/lsp/utils.vim]: check `l:winwidth > 0` if you do `let l:winwidth -= 11`.

Comment (smhc) [reply autoload/lsp/utils.vim]: Added a check for > 5
If the width is too small, it just outputs it non-truncated and you get the prompt. There's only so much you can do when the width doesn't accommodate the message - leave it up to vim to handle.

File: _config.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+theme: jekyll-theme-merlot
```
Comment (mattn) [inline _config.yml]: Is this required?

**Comment by mattn** [comment]
Please squash commits.

**Comment by prabirshrestha** [comment]
Squashed using github ui and merged. Thanks.

This has been one my long outstanding annoyance with vim-lsp too. :)

**Pull request merged**